restli-tools/src/main/java/com/linkedin/restli/tools/compatibility/ResourceCompatibilityChecker.java

```diff
@@ -1382,4 +1389,73 @@ private void loadResourceLevelServiceErrors(RecordTemplate previousRecord, Recor
       _currResourceLevelErrors = new ServiceErrorsSchema(currentRecord.data()).getServiceErrors(GetMode.DEFAULT);
     }
   }
+
+  /**
+   * Checks the compatibility of max batch size.
+   *
+   * @param prevMaxBatchSize previous max batch size
+   * @param currMaxBatchSize current max batch size
+   */
+  private void checkMaxBatchSizeAnnotation(MaxBatchSizeSchema prevMaxBatchSize, MaxBatchSizeSchema currMaxBatchSize)
+  {
+    if (prevMaxBatchSize == currMaxBatchSize)
+    {
+      return;
+    }
+
+    if (prevMaxBatchSize == null)
+    {
+      // Adding MaxBatchSize
+      if (currMaxBatchSize.isValidate())
+      {
+        // Adding MaxBatchSize with validation on
+        _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_ADDED_WITH_VALIDATION_ON, _infoPath);
+      }
+      else
+      {
+        // Adding MaxBatchSize with validation off
+        _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_ADDED_WITH_VALIDATION_OFF, _infoPath);
+      }
+    }
+    else if (currMaxBatchSize == null)
+    {
+      // Removing MaxBatchSize
+      _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_REMOVED, _infoPath);
+    }
+    else
+    {
+      int prevValue = prevMaxBatchSize.getValue();
+      int currValue = currMaxBatchSize.getValue();
+      boolean prevValidate = prevMaxBatchSize.isValidate();
+      boolean currValidate = currMaxBatchSize.isValidate();
+      if(prevValue < currValue)
+      {
+        // Increasing max batch size value
+        _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_VALUE_INCREASED, _infoPath);
+      }
+      else if (prevValue == currValue)
+      {
+        if (prevValidate && !currValidate)
+        {
+          _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_VALUE_TURN_OFF_VALIDATION, _infoPath);
+        }
+        else if (!prevValidate && currValidate)
+        {
+          _infoMap.addRestSpecInfo(CompatibilityInfo.Type.MAX_BATCH_SIZE_VALUE_TURN_ON_VALIDATION, _infoPath);
+        }
```

Comment:
This should be the highest precedence check

Reply:
Updated